fix(MediaPage): Links to ignore non-string actions

JoshuaCarter opened the pull request:
When a link is constructed via `ContentController::Link()` it passes `true` as the action to:

> force the expanded link to be returned so that form methods and similar will function properly

In this case it's slapping '/1/' onto links that are invoked via a template (because true = 1).

https://github.com/silverstripe/silverstripe-cms/blob/4/code/Controllers/ContentController.php#L95-L105

nglasl commented on src/pages/MediaPage.php:
```diff
@@ -413,7 +413,7 @@ public function Link($action = null) {
 			"{$date}{$this->URLSegment}/"
 		);
-		if($action) {
+		if(is_string($action) && !empty($action)) {
```
Just personal preference, but could this be `if($action && is_string($action))`? This will also need the same change for `AbsoluteLink` further below to ensure that doesn't hit a similar issue at some point. Nice pick up :smile:

JoshuaCarter replied:
Ah yep, done.

Commit:
fix(MediaPage): Links to ignore non-string actions

When a link is constructed via ContentController::Link() it passes
'true' as the action to "force the expanded link to be returned so that
form methods and similar will function properly". In this case it's
slapping '/1/' onto link that are invoked via a template (because true =
1).
